File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/RemoveEntityEffectPacket.java
package gg.mineral.server.network.packet.play.clientbound;

import gg.mineral.api.network.packet.Packet;
import io.netty.buffer.ByteBuf;

public final record RemoveEntityEffectPacket(int entityId, byte effectId) implements Packet.OUTGOING {
    @Override
    public void serialize(ByteBuf os) {
        os.writeInt(entityId);
        os.writeByte(effectId);
    }

    @Override
    public byte getId() {
        return 0x1E;
    }
}

File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/RemoveEntityEffectPacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class RemoveEntityEffectPacket(val entityId: Int, val effectId: Byte) : Packet.OUTGOING {
    override fun serialize(os: ByteBuf) {
        os.writeInt(entityId)
        os.writeByte(effectId.toInt())
    }

    override fun getId(): Byte {
        return 0x1E
    }
}
